src/shogun/machine/KernelMachine.cpp

```diff
@@ -38,13 +38,13 @@ struct S_THREAD_PARAM_KERNEL_MACHINE
 };
 #endif // DOXYGEN_SHOULD_SKIP_THIS
 
-KernelMachine::KernelMachine() : Machine()
+KernelMachine::KernelMachine() : NonParametricMachine()
 {
     init();
 }
 
 KernelMachine::KernelMachine(const std::shared_ptr<Kernel>& k, SGVector<float64_t> alphas,
-        SGVector<int32_t> svs, float64_t b) : Machine()
+        SGVector<int32_t> svs, float64_t b) : NonParametricMachine()
 {
     init();
```

Comment:
```suggestion
        SGVector<int32_t> svs, float64_t b) : KernelMachine()
{
```

Reply:
sorry, I don't understand why I should change `NonParametricMachine` to `KernelMachine`.

Reply:
https://www.variadic.xyz/2012/01/17/c11-delegating-constructors/